src/Compilers/CSharp/Portable/Syntax/SyntaxExtensions.cs

```diff
@@ -398,84 +400,115 @@ internal static bool IsOutVarDeclaration(this DeclarationExpressionSyntax p)
         /// <param name="argument">The argument that is passed to the action whenever it is invoked</param>
         internal static void VisitRankSpecifiers<TArg>(this TypeSyntax type, Action<ArrayRankSpecifierSyntax, TArg> action, in TArg argument)
```

Comment:
>VisitRankSpecifiers [](start = 29, length = 19)

Was this on the stack of the test that failed?

Reply:
Yes. This failed `EndToEndTests.DeeplyNestedGenerics`. This method was rather odd: it was written kinda iteratively, kinda recursively. Rather than half-measures, I decided the easiest (and most permanent) fix was to make it fully iterative with a manual stack.

The `VisitType` change was required by `EndToEndTests.Constraints`.

---
In reply to: [421223858](https://github.com/dotnet/roslyn/pull/44040#discussion_r421223858) [](ancestors = 421223858)